wallet/wallet.c

```diff
@@ -175,10 +172,7 @@ const struct utxo **wallet_select_coins(const tal_t *ctx, struct wallet *w,
 	if (satoshi_in < *fee_estimate + value) {
 		/* Could not collect enough inputs, cleanup and bail */
 		utxos = tal_free(utxos);
-		db_rollback_transaction(w->db);
```

Comment:
Wait, have we actually added the UTXO removal in the DB to the destructor? I wanted to avoid that on purpose since then a clean shutdown (which attempts to clean up `free`ing everything that was allocated) would result in us losing UTXOs from the database which we intended to keep around. Please don't mix memory management and DB persistence.